### ISSUE TITLE
Pyomo.DoE Development Plan

### ISSUE THREAD

**Issue by adowling2** [Pyomo/pyomo#2610]
The purpose of this meta issue is to track our development plans for Pyomo.DoE.

Immediate (November/December 2022):
- [x] Merge version 1 of Pyomo.DoE, #2294 
- [x] Merge Pyomo.DoE logo, #2527 

Pyomo.DoE Version 2 (start review in March 2023):
- [x] Remove extra index (using blocks)
- [x] Deprecate/remove sIpopt (pending discussion)
- [x] Update to use the TicTocTimer, see https://github.com/Pyomo/pyomo/pull/2294#discussion_r1024669360
- [x] Replace string options with `Enum`, see https://github.com/Pyomo/pyomo/pull/2294#discussion_r1024665131
- [x] Documentation and tutorial enhancements
- [x] Allow users to specify the names of the index sets (e.g., do not require time to be hard-coded as `t`, see https://github.com/Pyomo/pyomo/pull/2294#discussion_r1025732006, https://github.com/Pyomo/pyomo/pull/2294#discussion_r1025464771)

Pyomo.DoE Version 3+:
- [ ] Generalize measurement variance to consider correlations (not just diagonal)
- [ ] Add grey box objective function calculation
- [ ] Support decomposition with parapint
- [ ] Refactor interface for Pyomo.DoE and parmest to not define extra information (e.g., parameters, measurements) using strings, see https://github.com/Pyomo/pyomo/pull/2294#discussion_r1025732006

**Comment by adowling2** [comment]
Please assign this issue to @jialuw96 and @hglynch

**Comment by jialuw96** [comment]
- [ ] user define model time name https://github.com/Pyomo/pyomo/pull/2294#discussion_r1025464771

**Comment by jialuw96** [comment]
Pyomo.DoE Version 2 (Some other problems for discussion and enhancements, each one should be a small PR):
- [ ] Which is a better way to grab pyomo model variable values 
      - [ ] Get rid of `eval`;
      - [ ] Use `getattr` 
      - [ ] Use `ComponentUID` as in https://github.com/IDAES/idaes-pse/blob/274e58bef55f2f969f0df97cbb1fb7d99342388e/idaes/apps/uncertainty_propagation/sens.py#L296  
      - [ ] `find_component`  
 - [ ] How to flatten variables with multiple indexes, for e.g. m.C['CA', 'A1', 16,  2]? 
      - [ ] I saw the use of `UnindexedComponent_Set` and  `add_component`. Is this a good way?

**Comment by blnicho** [comment]
[ ] Improve test coverage (currently at ~66% but would prefer 85-100% coverage)

**Comment by jialuw96** [comment]
Note (To be updated)

https://github.com/Parapint/parapint/blob/fad622d912c03d7c23b70241d5d18b96942a4d4b/parapint/examples/burgers.py#L173

**Comment by adowling2** [comment]
Update documentation:
- [ ] @adowling2 careful read through
- [ ] Incorporate new Pyomo.DoE logo

**Comment by adowling2** [comment]
@blnicho @jsiirola Is Monday, Feb. 6 a reasonable cut-off date for minor PRs, e.g., updates to Pyomo.DoE documentation, before the next Pyomo release?

**Comment by blnicho** [comment]
@adowling2 yes Feb. 6 is fine for a cut-off date.

**Comment by adowling2** [comment]
@blnicho @jsiirola FYI, @jialuw96 and I are working on using blocks in Pyomo.DoE in https://github.com/jialuw96/pyomo/pull/15 which is a PR from one branch into @jialuw96's fork into a second branch on her fork. Once this is ready for review from the Pyomo team, we'll open PR into pyomo/main.

**Comment by jialuw96** [comment]
Pyomo.DoE version 3+: 
- [ ] add the integrator on an already discretized model.

**Comment by adowling2** [comment]
Pyomo.DoE version 3+:
- [ ] Add tests for `scenario.py` https://github.com/Pyomo/pyomo/pull/2794#discussion_r1200780723
- [ ] Add README file or top comment for each example https://github.com/Pyomo/pyomo/pull/2794#discussion_r1200769511
- [ ] Choose better function names to eliminate confusion between `compare` and `compute` https://github.com/Pyomo/pyomo/pull/2794#discussion_r1196803473

**Comment by djlaky** [comment]
@adowling2 @blnicho Any objections to closing this in favor of the new issue I made today (#3345)?
